processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -470,20 +480,71 @@ public String apply(@Nullable String input)
       }
     };
 
-    return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, null, rowMergerFn, indexSpec);
+    return makeIndexFiles(
+        indexes,
+        mergeAggregators(indexes),
+        outDir,
+        progress,
+        mergedDimensions,
+        mergedMetrics,
+        null,
+        rowMergerFn,
+        indexSpec
+    );
+  }
+
+  public static final AggregatorFactory[] mergeAggregators(List<IndexableAdapter> indexes)
```

Comment:
If we already do this somewhere else then let's keep the behavior and document it. What defines the order in which we pass segments to this function, is that documented?